### DISCUSSION TITLE
Prisma, Foreign Key Constraint violation on post

### DISCUSSION THREAD

**Discussion by TRed91** [prisma/prisma#25573] [Q&A] [answered]
### Question

Hello everyone! I got a problem with my API.
I'm using prisma for my db queries. 

I have two tables that reference each other twice

```
model user {
  userId   Int @id @default(autoincrement()) 
  userName String @unique
  email    String @unique
  password String
  about    String?

  sentMessages     message[] @relation(name: "sent")
  receivedMessages message[] @relation(name: "received")
  groups           group[]
}

model message {
  messageId   Int @id @default(autoincrement())
  messageText String
  timestamp   DateTime @default(now())

  authorId    Int
  author      user @relation(name: "sent", fields: [authorId], references: [userId])

  receiverId  Int?
  receiver    user? @relation(name: "received", fields: [receiverId], references: [userId])
}
```
when I send a post request to create a new message I receive a 'Foreign key constraint violated: `message_authorId_fkey (index)`' Error.

the query receives the authorId, receiverId and messageText

```
exports.createMassage = (authorId, receiverId, text) => {
    return prisma.message.create({
        data: {
            authorId: authorId,
            receiverId: receiverId,
            messageText: text
        },
    });
}
```
the controller passes proper values to the query. For example: 
authorId:2, receiverId:1 text:Hey John! It's me Jane. How are you doing?
Both id values are ints. I double checked.

This is the controller:

```
exports.createMessage = async(req, res) => {
    try {
        const receiver = await userDb.userGetByName(req.body.receiver);
        if (!receiver) {
            return res.status(400).json({ ok: false, data: null, message: 'Receiver not found.' })
        }
        const authorId = parseInt(req.params.userId);

        const result = await messageDb.createMassage(authorId, receiver.userId, req.body.message);
        return res.json({ ok: true, data: result });
    } catch (err) {
        console.error('Error creating message: ', err.message);
        return res.status(500).json({ ok: false, data: null, message: err.message });
    }
}
```

My guess is that I somehow messed up the double relation with the user model.

I'm still learning. I couldn't think of a better way to handle my db models for this case.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by TRed91** [comment, marked as answer]
My bad. It worked fine, my post request was sending the wrong id.

**Reply by nurul3101** [reply]
Glad to hear that you were able to resolve it.
